includes/core/functions.php

```diff
@@ -1601,6 +1601,64 @@ function wct_is_signup_allowed_for_current_blog() {
 	return apply_filters( 'wct_is_signup_allowed_for_current_blog', wct_allow_signups() );
 }
 
+/**
+ * Updates last_status post meta before saving talk.
+ *
+ * @since  1.2.0
+ */
+function wct_talks_update_last_status( $data, $postarr ) { 
+	// Checks if we are in 'talks' CPT and if not, abort
+	if( $data['post_type'] !== 'talks' ) :
+		return $data;
+	endif;
+	// Gets current talk status
+	$current_post_status = get_post_status( $postarr['ID'] );
+	// Updates last talk status meta
+	update_post_meta( $postarr['ID'], '_wct_talks_last_status', $current_post_status );
+	return $data;
+}
+
+/**
+ * Run a scheduled task to send emails for talks status updated the last 24h.
+ *
+ * @since  1.2.0
+ */
+function wct_talks_updated_talks_send_emails() {
+	// Gets all talks modified the last 24h
+	$args = array(
+		'post_type'			=> 'talks',
+		'post_status'		=> 'any',
+		'posts_per_page' 	=> -1,
+		'date_query' => array(
+			array(
+				'column' => 'post_modified_gmt',
+				'after'  => '1 day ago',
+			),
+		),
+	);
+	$query_talks = new WP_Query( $args );
+	if ( $query_talks->have_posts() ) :
+		while ( $query_talks->have_posts() ) :
+			$query_talks->the_post();
+			// Get talk infos
+			$talk_title = get_the_title();
+			$talk_author_email = get_the_author_meta( 'user_email' );
+			$last_status = get_post_meta( get_the_ID(), '_wct_talks_last_status', true );
+			$current_status = get_post_status( get_the_ID() );
+			// Checks if status has changed
+			if ( !empty( $last_status ) && $current_status != $last_status && $last_status != 'draft' ) : 
+				// Sends wp_mail to each author
+				wp_mail( 
+					$talk_author_email, 
+					'Status change for your talk "' . $talk_title . '"', 
+					'Your "' . $talk_title . '" talk status as changed from status "' . $last_status . '" to status "' . $current_status . '"'
+				);
+			endif;
+		endwhile;
+	endif;
+	wp_reset_postdata();
+}
```

Comment:
I think i'd do something like this (i haven't tested though!):

```php
// Get the Site Name only once.
$site_name = wp_specialchars_decode( get_option( 'blogname' ), ENT_QUOTES );

// Get Talk statuses only once.
$statuses = wct_get_statuses();

foreach ( $q['talks'] as $talk ) {
	$status      = get_post_status( $talk );
	$last_status = get_post_meta( $talk->ID, '_wct_talks_last_status', true );

	if ( ! $last_status || $status !== $last_status ) {
		continue;
	}

	// Remove the last status meta as we're about to notify the author.
	delete_post_meta( $talk->ID, '_wct_talks_last_status' );

	$author_email = wct_users_get_user_data( 'id', $talk->post_author, 'user_email' );

	// Avoid weird emails or statuses!
	if ( ! is_email( $author_email ) || ! isset( $statuses[ $status ] ) ) {
		continue;
	}

	wp_mail(
		$author_email,
		/* translators: 1: Site name */
		sprintf( __( '[%s] The status of your talk proposal has changed.', 'wordcamp-talks' ), $site_name ),
		/* translators: 1: linefeeds, 2: Talk title, 3: Talk status, 4: Talk permalink */
		sprintf(
			__( 'Hi there!%1$sThe status of your talk %2$s is now %3$s.%1$sLog in our website to view it: %4$s', 'wordcamp-talks' ),
			"\n\n",
			esc_html( $talk->post_title ),
			esc_html( $statuses[ $status ] ),
			esc_url( wct_talks_get_talk_permalink( $talk ) )
		)
	);
}
```

The idea is to use the post meta like a marker to notify the author. I'm not sure the author is interested of knowing the previous status of his talk, he's probably more interested to know if it has been selected, rejected or "inserted" into a backup list.

In this last possibility he can set his talk as "rejected" if he doesn't want to be a backup plan, reason why i think it's important to include the link to the Talk Proposal.

Reply:
Looks fine to me 👍

Reply:
Great ! how do you want to proceed ? Do you want to edit your PR accordingly to what we agreed on or do you want me to edit master directly ?